### ISSUE TITLE
In WPF, filter does not work

### ISSUE THREAD

**Issue by zerodev1200** [Cysharp/ObservableCollections#64]
# Filter Not Working in WPF Since v2.2.0

## Issue Description
Since `v2.2.0`, the filter has stopped working in WPF. We believe the cause is that `IList` in `ListNotifyCollectionChangedSynchronizedView<T, TView>` cannot properly retrieve elements after filtering.

## Partial Fix and New Problem
By modifying parts of the code in `v2.2.2`, filtering now functions. However, after filtering, adding items to the original source causes a `System.InvalidOperationException`.

## Reproduction
This issue can be reproduced in [this repository](https://github.com/zerodev1200/ObservableCollections_Filter).

## Code Changes
In `NotifyCollectionChangedSynchronizedView<T, TView>`:
```csharp
public virtual int Count => parent.Count; //added virtual keyword
```

In `ListNotifyCollectionChangedSynchronizedView<T, TView>`:
```csharp
public TView this[int index]
{
    get
    {
        lock (view.SyncRoot)
        {
            //return view.list[index].Item2;    // Old implementation
            return view.ElementAt(index).View;  // New implementation
        }
    }
    set => throw new NotSupportedException();
}

//added
public override int Count
{
    get
    {
        lock (view.SyncRoot)
        {
            //view.Count always returns the value before filtering 
            return view.Count();
        }
    }
}
```

## Observation
In versions prior to `v2.1.4`, where `IList` was not implemented, there was no access to the view during filtering or when adding elements to the original list (at least not during debugging).

However, in Views that implement `IList`, these operations cause access to occur, leading to display issues and exceptions.

**Comment by neuecc** [comment]
Thank you, I am currently working on a large-scale refactoring, and those issues should be resolved.
However, it will also involve a lot of breaking changes.
https://github.com/Cysharp/ObservableCollections/pull/65

**Comment by zerodev1200** [comment]
Thank you for your hard work.
After testing with this code, I found that filtering works perfectly, and furthermore, adding and removing values in this state can be done without any issues.
However, there was still a challenging problem with `IList`.
When using range methods like `AddRange()` or `RemoveRange()` on the sourceList, the following exception occurs:
Location: `System.Windows.Data.ListCollectionView.ValidateCollectionChangedEventArgs()`
Type: `System.NotSupportedException: 'Range actions are not supported.'`
Looking at the relevant code, it appears as follows:
```csharp
if (e.NewItems.Count != 1)
{
    throw new NotSupportedException(SR.RangeActionsNotSupported);
}
```
I personally don't use `***Range()` methods, but I happened to discover this, so I'm reporting it to you.
